class.jetpack-client.php

```diff
@@ -65,10 +65,18 @@ public static function remote_request( $args, $body = null ) {
 		if ( is_null( $body ) ) {
 			$body_hash = '';
 		} else {
-			if ( !is_string( $body ) ) {
+
+			// Allow arrays to be used in passing data.
+			$body_to_hash = $body;
+			if ( is_array( $body ) ) {
+				$body_to_hash = json_encode( self::_stringify_data( $body ) );
```

Comment:
aha, well, that is there because it needs to turn data into strings so that they match the strings that will appear on the wpcom side, when the POST is made. HTTP requests have no concept of types like int or obj, so I'm turning things into their base types. This was explained in the method description for `_stringify_data`